src/radical/pilot/utils/prof_utils.py

```diff
@@ -199,6 +199,114 @@
+                                      {ru.EVENT: 'staging_in_start' , ru.STATE: None                       }  ],
+        'p_pmngr_stage_in'        : [ {ru.EVENT: 'staging_in_start' , ru.STATE: None                       }, 
+                                      {ru.EVENT: 'staging_in_stop'  , ru.STATE: None                       }  ], 
+        'p_pmngr_submission_init' : [ {ru.EVENT: 'staging_in_stop'  , ru.STATE: None                       }, 
+                                      {ru.EVENT: 'submission_start' , ru.STATE: None                       }  ], 
+        'p_pmngr_submission'      : [ {ru.EVENT: 'submission_start' , ru.STATE: None                       },  
+                                      {ru.EVENT: 'submission_stop'  , ru.STATE: None                       }  ],
+        'p_pmngr_scheduling_init' : [ {ru.EVENT: 'submission_stop'  , ru.STATE: None                       },  
+                                      {ru.EVENT: 'state'            , ru.STATE: rps.PMGR_ACTIVE_PENDING    }  ], 
+        'p_pmngr_scheduling'      : [ {ru.EVENT: 'state'            , ru.STATE: rps.PMGR_ACTIVE_PENDING    },     # batch system queue time
+                                      {ru.EVENT: 'bootstrap_0_start', ru.STATE: None                       }  ],
+        'p_agent_ve_setup_init'   : [ {ru.EVENT: 'bootstrap_0_start', ru.STATE: None                       },  
+                                      {ru.EVENT: 've_setup_start'   , ru.STATE: None                       }  ], 
+        'p_agent_ve_setup'        : [ {ru.EVENT: 've_setup_start'   , ru.STATE: None                       },  
+                                      {ru.EVENT: 've_setup_stop'    , ru.STATE: None                       }  ], 
+        'p_agent_ve_activate_init': [ {ru.EVENT: 've_setup_stop'    , ru.STATE: None                       },  
+                                      {ru.EVENT: 've_activate_start', ru.STATE: None                       }  ], 
+        'p_agent_ve_activate'     : [ {ru.EVENT: 've_activate_start', ru.STATE: None                       },
+                                      {ru.EVENT: 've_activate_stop' , ru.STATE: None                       }  ],
+        'p_agent_install_init'    : [ {ru.EVENT: 've_activate_stop' , ru.STATE: None                       },
+                                      {ru.EVENT: 'rp_install_start' , ru.STATE: None                       }  ],
+        'p_agent_install'         : [ {ru.EVENT: 'rp_install_start' , ru.STATE: None                       },
+                                      {ru.EVENT: 'rp_install_stop'  , ru.STATE: None                       } ],
+        'p_agent_launching'       : [ {ru.EVENT: 'rp_install_stop'  , ru.STATE: None                       },
+                                      {ru.EVENT: 'state'            , ru.STATE: rps.PMGR_ACTIVE            }  ],
+        'p_agent_runtime'         : [ {ru.EVENT: 'state'            , ru.STATE: rps.PMGR_ACTIVE            },     # total pilot runtime
```

Comment:
We can add the timestamp where the pilot receives it's termination command - specifically for EnTK this may be relevant as that command arrives rather late sometimes.

Reply:
I derived the events you see from an actual run. Some events might have been missing (bootstrap_0_stop was missing). What would be the event pair? Would those be contiguous?

Reply:
Have a look [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/utils/prof_utils.py#L36) where we define the current durations used for RA utilization: the `ignore` and `term` durations would be the ones to check.  Yes, they are continuous.

Reply:
Apologies, I had missed this one. I pushed the termination events following, I think, your indications. How should we managed the 'tagging' of the events? I am not sure I understand why we would ignore an event, and do we know for certain that when we record `bootstrap_0_stop` the pilot has been already terminated? This is why we assume that `bootstrap_0_stop -> DONE/CANCELED/FAILED` does not consume resources?

Reply:
`ignore` here means that the respective duration is ignored when computing utilization.  `bootstrap_0_stop` is the last event written by the agent (or, more precisely, by the bootstrapper which started the agent) [before it finishes](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/agent/bootstrap_0.sh#L1928).  Well, it does some more things, like creating a tarball of the profiles - but at that point the profiles are closed and we consider the agent done.

Reply:
Taggin like in a sort of annotation of groups of duration, i.e., ignore, consumed, etc.  For me, those are cumbersome as they are meaningful only in the contest of our current implementation of resource utilization. In all other contexts, I need to get rid of them. If I have to use different 'tags' also in the *DEBUG data structures, then RA users will not be able to directly use `rp.utils.*DEBUG`. I will have to go around it with a dedicated call that eliminate  one of the level of the dict of dict, or something like that.

Reply:
About pilot runtime, I guess bootstrap_0_start/stop is close enough? Do you have a better suggestion?

Reply:
Tagging: I see.  I would suggest to do that in a separate structure.  For example, the utilization could be defined as:
```py
pilot_utilization = {
    'provide' : {
        'total': ['p_agent_runtime']
    },
    'consume': {
        'boot'  : ['p_agent_ve_setup_init', 'p_agent_ve_setup', ...,]
        'launch': ['p_agent_launching'],
        'term'  : ['p_agent_terminate']
    },
    'agent': {
        'total': ['p_agent_runtime']
    }
}
```
So we would list all durations which contribute to a particular metric.  Would that make sense to you?

Reply:
> About pilot runtime, I guess bootstrap_0_start/stop is close enough? Do you have a better suggestion?

This one is exactly correct IMHO.

Reply:
OK, I gave it a go and I come up with something similar to what you seemed to suggest. Please have a look and see whether it matches your expectations. To be usable, it would require extending the RU methods but I would do that later on as, at the moment, nobody uses the DEBUG durations for RU.

Reply:
Looks good to me!  I have some minor comments, but they do not hold up a merge.